### PR TITLE
MGDSTRM-8995: implement feature flag for topic config policy

### PR DESCRIPTION
**Pull request by showuon**
What I did in this PR:
1. add a `kas.policy.topic-config.topic-config-policy-enforced` config, and default to false
2. disable all topic config rules when `kas.policy.topic-config.topic-config-policy-enforced` is false
3. add tests

### CODE REVIEW

**Comment by showuon** [comment]
@grdryn @SamBarker , since we are waiting for the topic config policy feature flag for the new release. Implement it first. Thanks.

File: src/main/java/io/bf2/kafka/common/Config.java
Comment (SamBarker) [inline src/main/java/io/bf2/kafka/common/Config.java:128]: The point of the new flag is to be a switch to the "original" create topic policy or this new one. So while a boolean flag could work I don't think it should be called `topic-config-policy-enforced`.

One other option could be to make it a string `$ALTER_CONFIG_POLICY_PREFIX.version`.
`version=original` the old behaviour
`version=MGDSTRM-8524` the behaviour from this epic
`version=MGDSTRM-????` switch to the behaviour of next version. 

But then I'm not really keen on the feature flags being long lived things.

**Review by SamBarker — CHANGES_REQUESTED** [review]
The feature flag is intended to be a toggle between the hold behaviour of the plugin and this update not a mechanism to turn the policy off.

**Comment by showuon** [comment]
> The feature flag is intended to be a toggle between the hold behaviour of the plugin and this update not a mechanism to turn the policy off.

@SamBarker , thanks for the comment. But I didn't get your idea here. Did you mean the config name should not be a boolean only? Or the boolean thing is just another comment? What do you mean not to turn the policy off? We want to disable the policy if disabled, don't we? 

My thought is we add this feature flag config temporarily, and will be removed after everything else is merged and not blocking with each other. So the name should not be important. But with your `version` proposal, it's a good suggestion. It will be a long-lived config that allows future gating use, right?

**Comment by SamBarker** [comment]
> My thought is we add this feature flag config temporarily, and will be removed after everything else is merged and not blocking with each other. So the name should not be important. But with your version proposal, it's a good suggestion. It will be a long-lived config that allows future gating use, right?

Yes. The name is somewhat moot if short lived. I think my thoughts around version are just muddying the waters. I don't see any real need for a long lived flag.

> We want to disable the policy if disabled, don't we?

No, at least in my head the feature flag is about switching from the previous version to the new version not tooling to turn the policy off entirely.

**Comment by showuon** [comment]
> No, at least in my head the feature flag is about switching from the previous version to the new version not tooling to turn the policy off entirely.

Yes, I agree. But in our case, the previous version means no topic policy. So switching back to previous version means turn the policy off. I don't think we are on the same page. We can discuss it in standup tomorrow. :)

**Review by SamBarker — APPROVED** [review]
Resolved my misunderstand. The flag in this PR only covers the behaviour introduced in [MGDSTRM-8524](https://issues.redhat.com/browse/MGDSTRM-8524)

The details of what a version flag imply are loosely defined and I don't think they are worth pursuing at this stage.

Thus this PR LGTM :shipit:

**Comment by showuon** [comment]
Thanks, Sam!

**Review by k-wall — APPROVED** [review]

**Comment by showuon** [comment]
Thanks for the review, Sam, Keith!

**Pull request merged**
